homeassistant/components/mammotion/lawn_mower.py

```diff
@@ -0,0 +1,321 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MammotionConfigEntry, MammotionReportUpdateCoordinator
+from .const import COMMAND_EXCEPTIONS, DOMAIN, LOGGER
+from .entity import MammotionBaseEntity
+
+SERVICE_START_MOWING = "start_mow"
+SERVICE_CANCEL_JOB = "cancel_job"
+
+START_MOW_SCHEMA = {
+    vol.Optional("is_mow", default=True): cv.boolean,
+    vol.Optional("is_dump", default=True): cv.boolean,
+    vol.Optional("is_edge", default=False): cv.boolean,
+    vol.Optional("collect_grass_frequency", default=10): vol.All(
+        vol.Coerce(int), vol.Range(min=5, max=100)
+    ),
+    vol.Optional("border_mode", default=1): vol.In([0, 1]),
+    vol.Optional("job_version", default=0): vol.Coerce(int),
+    vol.Optional("job_id", default=0): vol.Coerce(int),
+    vol.Optional("speed", default=0.3): vol.All(
+        vol.Coerce(float), vol.Range(min=0.2, max=1.2)
+    ),
+    vol.Optional("ultra_wave", default=2): vol.In([0, 1, 2, 10, 11]),
+    vol.Optional("channel_mode", default=0): vol.In([0, 1, 2, 3]),
+    vol.Optional("channel_width", default=25): vol.All(
+        vol.Coerce(int), vol.Range(min=5, max=35)
+    ),
+    vol.Optional("rain_tactics", default=1): vol.In([0, 1]),
+    vol.Optional("blade_height", default=25): vol.All(
+        vol.Coerce(int), vol.Range(min=15, max=100)
+    ),
+    vol.Optional("toward", default=0): vol.All(
+        vol.Coerce(int), vol.Range(min=-180, max=180)
+    ),
+    vol.Optional("toward_included_angle", default=0): vol.All(
+        vol.Coerce(int), vol.Range(min=-180, max=180)
+    ),
+    vol.Optional("toward_mode", default=0): vol.In([0, 1, 2]),
+    vol.Optional("mowing_laps", default=1): vol.In([0, 1, 2, 3, 4]),
+    vol.Optional("obstacle_laps", default=1): vol.In([0, 1, 2, 3, 4]),
+    vol.Optional("start_progress", default=0): vol.All(
+        vol.Coerce(int), vol.Range(min=0, max=100)
+    ),
+    vol.Required("areas"): vol.All(
+        cv.ensure_list, [cv.entity_id]
+    ),  # This assumes `areas` are entity IDs from the integration
+}
+
+
+def get_entity_attribute(
+    hass: HomeAssistant, entity_id: str, attribute_name: str
+) -> str | None:
+    """Get an attribute from an entity."""
+    # Get the state object of the entity
+    entity = hass.states.get(entity_id)
+
+    # Check if the entity exists and has attributes
+    if entity and attribute_name in entity.attributes:
+        # Return the specific attribute
+        return entity.attributes.get(attribute_name, None)
+    # Return None if the entity or attribute does not exist
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MammotionConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Luba config entry."""
+    mammotion_devices = entry.runtime_data
+
+    for mower in mammotion_devices:
+        async_add_entities([MammotionLawnMowerEntity(mower.reporting_coordinator)])
```

Comment:
We should run async add entities as little as possible, so ideally just use a list comprehension inside of the list passed in